napari/_vispy/vispy_axes_visual.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+from vispy.scene.visuals import Line
+from vispy.visuals.transforms import STTransform
+
+from ..components._viewer_constants import AxesStyle
+from ..utils.colormaps.standardize_color import transform_color
+
+
+def make_dashed_line(num_dashes, axis):
+    """Make a dashed line.
+
+    Parameters
+    ----------
+    num_dashes : int
+        Numbers of dashes in the line.
+    axis :
+        Axis which is dashed.
+
+    Returns
+    -------
+    np.ndarray
+        Dashed line.
+    """
+    dashes = np.linspace(0, 1, num_dashes * 2)
+    dashed_line_ends = np.concatenate(
+        [[dashes[2 * i], dashes[2 * i + 1]] for i in range(num_dashes)], axis=0
+    )
+    dashed_line = np.zeros((2 * num_dashes, 3))
+    dashed_line[:, axis] = np.array(dashed_line_ends)
+    return dashed_line
+
+
+class VispyAxesVisual:
+    """Axes indicating world coordinate origin and orientation.
+
+    Axes are colored x=cyan, y=yellow, z=magenta or dashed with
+    x=solid, y=dotted, z=dashed, depending on styling.
```

Comment:
Ah, here's the docstring.

I think "depending on styling" is very vague, and doesn't give much of a clue about how to use it.

Reply:
this language has now been changed, agreed was vague